workers/gc/gcworker.py

```diff
@@ -39,13 +49,31 @@ def _garbage_collection_repos(self):
             assert features.GARBAGE_COLLECTION
 
             try:
-                repository = Repository.get(id=repo_ref.id)
-            except Repository.DoesNotExist:
-                return
+                with GlobalLock(
+                    "REPO_GARBAGE_COLLECTION_%s" % repo_ref.id, lock_ttl=REPOSITORY_GC_TIMEOUT + 60,
+                ) if not skip_lock_for_testing else empty_context():
```

Comment:
I didn't know you could put conditionals in `with`, neat.

Reply:
Yeah, I discovered it recently.